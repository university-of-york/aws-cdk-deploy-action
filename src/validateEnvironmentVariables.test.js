const validateEnvironmentVariables = require('./validateEnvironmentVariables');

test('should throw if any of the required environment variables are missing', () => {
    try {
        validateEnvironmentVariables();
    } catch (error) {
        expect(error.toString()).toMatchSnapshot();
    }
});

test('should return an object with required environment variables', () => {
    process.env.AWS_ACCOUNT_ID = '1234';
    process.env.AWS_ACCESS_KEY_ID = 'account_id';
    process.env.AWS_SECRET_ACCESS_KEY = 'a_password';

    expect(validateEnvironmentVariables()).toMatchSnapshot();
});
