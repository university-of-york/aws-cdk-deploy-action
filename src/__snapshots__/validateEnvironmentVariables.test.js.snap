// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return an object with required environment variables 1`] = `
Object {
  "AWS_ACCESS_KEY_ID": "account_id",
  "AWS_ACCOUNT_ID": "1234",
  "AWS_SECRET_ACCESS_KEY": "a_password",
}
`;

exports[`should throw if any of the required environment variables are missing 1`] = `
"Error: Could not find value(s) in environment for: 
AWS_ACCOUNT_ID
AWS_ACCESS_KEY_ID
AWS_SECRET_ACCESS_KEY"
`;
